lib/linked_list.rb

```diff
@@ -18,120 +18,254 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: o(1)
+    # Space Complexity: o(1)
     def add_first(value)
-      raise NotImplementedError
+  
+      new_node = Node.new(value)
+      
+      if @head != nil?
+        new_node.next = @head
+      end
+
+      # Set the head to become the new node
+      @head = new_node

```

Comment:
A bit more condensed. 

```suggestion
    @head = Node.new(value, @head)
```